.github/workflows/klocwork-pull-request.yml

```diff
@@ -81,13 +89,37 @@ jobs:
         run: kwinject --output kwinject.out ninja -C build -v -k0
 
       - name: incrementally analyze build of head ref
-        run: kwcheck run --build-spec kwinject.out -F scriptable --report issues.txt --project-dir /home/klocwork/.kwlp --settings-dir /home/klocwork/.kwps
+        run: kwcheck run --build-spec kwinject.out -F scriptable --report /home/klocwork/child_issues.txt --project-dir /home/klocwork/.kwlp --settings-dir /home/klocwork/.kwps
+
+      - name: get detailed child issue
+        run: kwcheck list -F detailed --project-dir /home/klocwork/.kwlp --settings-dir /home/klocwork/.kwps | head -n -3 > /home/klocwork/child_issues_details.txt
+
+      - name: upload klockwork issues
+        uses: actions/upload-artifact@v2
+        with:
+          name: klocwork-${{ matrix.image }}-${{ matrix.config }}-${{ github.run_id }}
+          path: /home/klocwork/child_issues_details.txt
+          if-no-files-found: error
+      
+      - name: get child issue ids
+        run: awk -F ';' '{print $1}' /home/klocwork/child_issues.txt | sort > /home/klocwork/child_issues_ids.txt
+      
+      - name: fixed issues
+        run: |
+          FIXED_ISSUE_ID=$(comm -13 /home/klocwork/child_issues_ids.txt /home/klocwork/parent_issues_ids.txt | paste -sd'|')
+          if [ -n "${FIXED_ISSUE_ID}" ]
+          then
+            awk -vRS='----------' "/(${FIXED_ISSUE_ID}) \(Local\)/{print}" /home/klocwork/parent_issues_details.txt
+          fi
 
       - name: ensure no new issues have been found in head ref
         run: |
-          kwcheck list -F detailed --project-dir /home/klocwork/.kwlp --settings-dir /home/klocwork/.kwps
-          size=$(stat --printf=%s issues.txt)
-          test "$size" = "0"
+          NEW_ISSUE_ID=$(comm -13 /home/klocwork/parent_issues_ids.txt /home/klocwork/child_issues_ids.txt | paste -sd'|')
+          if [ -n "${NEW_ISSUE_ID}" ]
+          then
+            awk -vRS='----------' "/(${NEW_ISSUE_ID}) \(Local\)/{print}" /home/klocwork/child_issues_details.txt
```

Comment:
Why are the input records separated by
```
---------------------------------------------------------------------------
```
whereas the output records are separated by
```
-----
```
and the record separator is `----------`?

`(^|\n)-{75}\n` should match the record separator exactly, e.g.:
```
awk -vRS='(^|\n)-{75}\n' "/^(1|23) \(Local\)/{print}" child_issues_details.txt
```
```
1 (Local) /__w/fpga-runtime-for-opencl/fpga-runtime-for-opencl/test/acl_svm_test.cpp:382 NPD.FUNC.MUST (1:Critical) Analyze
Pointer 'src_ptr' returned from call to function 'clSVMAlloc' at line 371 may be NULL and will be dereferenced at line 382.
  * acl_svm_test.cpp:372: 'src_ptr' is assigned the return value from function 'clSVMAlloc'.
    * acl_svm.cpp:171: 'clSVMAllocIntelFPGA(context, flags, size, alignment)' is assigned the return value from function 'clSVMAllocIntelFPGA'.
      * acl_svm.cpp:51: 'clSVMAllocIntelFPGA' explicitly returns a NULL value.
    * acl_svm.cpp:171: 'clSVMAllocIntelFPGA(context, flags, size, alignment)' is returned by 'clSVMAlloc'.
  * acl_svm_test.cpp:382: 'src_ptr' is explicitly dereferenced.
Current status 'Analyze'

23 (Local) /__w/fpga-runtime-for-opencl/fpga-runtime-for-opencl/lib/pkg_editor/src/pkg_editor.c:278 NPD.FUNC.MIGHT (1:Critical) Analyze
Pointer 'data' returned from call to function 'get_name_data_ptr' at line 269 may be NULL and may be dereferenced at line 278.
  * pkg_editor.c:269: 'data' is assigned the return value from function 'get_name_data_ptr'.
    * pkg_editor.c:133: 'get_name_data_ptr' explicitly returns a NULL value.
  * pkg_editor.c:278: 'data' is explicitly dereferenced.
Current status 'Analyze'
```

Reply:
I like your creative use of `awk` :shell: :ninja:, but I am unsure whether it is a good idea to parse the `detailed` output at all, since that is what `scriptable` (or `json`) output is intended for. [`kwcheck list`](https://docs.roguewave.com/en/klocwork/current/kwcheck#kwchecklist) has an `--id` option that allows specifying a comma-separated list of issues. Would that satisfy your goal to list a subset of the issues?

Reply:
The caveat with using kwcheck, which I realized when I was testing the flow is: When the issue is resolved in the child build, the issue will be suppressed (although its issue id is preserved). For example there were original issue 1-5, and issue 4 was resolved, but a new issue was also discovered, then the resulting child issues will be 1,2,3,5,6. This means if we try to list the fixed issues (4), then this will not show.

Reply:
> awk -vRS='(^|\n)-{75}\n' "/^(1|23) \(Local\)/{print}" child_issues_details.txt

Thank you, yes this works.

Reply:
Thanks @sherry-yuan, I verified that even
```
kwcheck set-status --status Analyze 123
kwcheck list --id 123
```
does not list a resolved issue.

In that case, what do you think of having separate paths for the parent and child Klocwork projects, and copying the parent to the child project before the child analysis? The current solution fails silently when the `detailed` output format changes in a new Klocwork version, which should be expected since it is not intended for parsing.

Reply:
Thank you! This is a great idea and it worked 🎉